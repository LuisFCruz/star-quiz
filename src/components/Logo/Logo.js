import React, { Component } from 'react'
import PropTypes from 'prop-types'

export class Logo extends Component {
  static propTypes = {
    image: PropTypes.string.isRequired,
    className: PropTypes.string
  }

  render() {
    const { image, className = '' } = this.props;
    return (
      <img
        src={image}
        className={className}
        alt=""
      />
    )
  }
}

export default Logo;
