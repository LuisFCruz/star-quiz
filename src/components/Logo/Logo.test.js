import React from 'react';
import renderer from 'react-test-renderer';

import { Logo } from './Logo';

describe('<Logo />', () => {
  
  test('render without className', () => {
    const props = {
      image: 'http://star-wars.com/image.jpg',
    };
    const wrapper = renderer.create(<Logo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render with className', () => {
    const props = {
      image: 'http://star-wars.com/image.jpg',
      className: 'something'
    };
    const wrapper = renderer.create(<Logo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
