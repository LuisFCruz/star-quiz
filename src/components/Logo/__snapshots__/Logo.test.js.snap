// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> render with className 1`] = `
<img
  alt=""
  className="something"
  src="http://star-wars.com/image.jpg"
/>
`;

exports[`<Logo /> render without className 1`] = `
<img
  alt=""
  className=""
  src="http://star-wars.com/image.jpg"
/>
`;
